modules/Router.js

```diff
@@ -91,7 +90,7 @@ const Router = React.createClass({
       } else {
         // Keep the identity of this.router because of a caveat in ContextUtils:
         // they only work if the object identity is preserved.
-        assign(this.router, this.createRouterObject(state))
+        updateRouterObjectState(this.router, state)
```

Comment:
Or call it something like `assignRouterState` and use it in `createRouterObject` as well.